src/msgpack/encoder.rs

```diff
@@ -254,7 +254,7 @@ pub fn pack_bool(buf: &mut Option<&mut Buffer>, val: bool) -> Result<usize> {
 pub fn pack_map_begin(buf: &mut Option<&mut Buffer>, length: usize) -> Result<usize> {
     match length {
         val if val < 16 => pack_half_byte(buf, 0x80 | (length as u8)),
-        val if val >= 16 && val < 2 ^ 16 => pack_i16(buf, 0xde, length as i16),
+        val if (16..1 << 16).contains(&val) => pack_i16(buf, 0xde, length as i16),
```

Comment:
Adding an extra set of braces might make this a bit easier to read; the `16..1` part looks quite odd at first, otherwise:

```suggestion
        val if (16..(1 << 16)).contains(&val) => pack_i16(buf, 0xde, length as i16),
```

Though I'm not sure if rustfmt would just remove the unnecessary braces again anyway? (Or maybe clippy complains?)

Reply:
Done